dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -299,8 +301,11 @@ public int positionedRead(long pos, byte[] b, int off, int len) throws IOExcepti
           ReadType.READ_INTO_BYTE_ARRAY, pos, true);
     } catch (IOException | RuntimeException e) {
       LOG.warn("Failed to read from Alluxio's page cache.", e);
-      MetricsSystem.counter(MetricKey.CLIENT_CACHE_POSITION_READ_FALLBACK.getName()).inc();
-      return getExternalFileInStream().positionedRead(pos, b, off, len);
+      if (mFallbackEnabled) {
+        MetricsSystem.counter(MetricKey.CLIENT_CACHE_POSITION_READ_FALLBACK.getName()).inc();
+        return getExternalFileInStream().positionedRead(pos, b, off, len);
+      }
```

Comment:
shall we make the fallback in `readInternal` rather than `positionedRead` ?
today, it looks to me we only fallback on `positionedRead` not in the other two read methods

Reply:
it might be difficult to reuse the logic, because it will call different method when fallback